metals/src/main/scala/scala/meta/internal/metals/Indexer.scala

```diff
@@ -46,8 +45,7 @@ final case class Indexer(
     timerProvider: TimerProvider,
     scalafixProvider: () => ScalafixProvider,
     indexingPromise: Promise[Unit],
-    ammonite: () => Ammonite,
-    lastImportedBuilds: () => List[ImportedBuild],
+    buildData: () => Seq[(String, BuildTargets.WritableData, ImportedBuild)],
```

Comment:
This is pretty hard to grok at first sight, especially when you create the `Indexer` with
```scala
    () =>
      Seq(
        (
          "main",
          mainBuildTargetsData,
          ImportedBuild.fromList(lastImportedBuilds)
        ),
        ("ammonite", ammonite.buildTargetsData, ammonite.lastImportedBuild)
      ),
```

Would it be possible to instead having a tuple here to create some sort of case class or something for this?

Reply:
Sure!

Reply:
Done.